diff --git a/uswds_paragraphs.module b/uswds_paragraphs.module
index 212d71e..864fd82 100644
--- a/uswds_paragraphs.module
+++ b/uswds_paragraphs.module
@@ -61,50 +61,50 @@ function uswds_paragraphs_preprocess_paragraph(&$variables) {
  */
 function _uswds_paragraphs_fraction_to_class($numerator, $denominator) {
   if ($numerator == $denominator) {
-    return 'usa-width-one-whole';
+    return 'tablet:grid-col-12';
   }
   // We could do a lot of fancy math here. But we won't.
   $classes = [
     // Denominators first.
     2 => [
       // Then numerators.
-      1 => 'usa-width-one-half',
+      1 => 'tablet:grid-col-6',
     ],
     3 => [
-      1 => 'usa-width-one-third',
-      2 => 'usa-width-two-thirds',
+      1 => 'tablet:grid-col-4',
+      2 => 'tablet:grid-col-8',
     ],
     4 => [
-      1 => 'usa-width-one-fourth',
-      2 => 'usa-width-one-half',
-      3 => 'usa-width-three-fourths',
+      1 => 'tablet:grid-col-3',
+      2 => 'tablet:grid-col-6',
+      3 => 'tablet:grid-col-9',
     ],
     6 => [
-      1 => 'usa-width-one-sixth',
-      2 => 'usa-width-one-third',
-      3 => 'usa-width-one-half',
-      4 => 'usa-width-two-thirds',
-      5 => 'usa-width-five-sixths',
+      1 => 'tablet:grid-col-2',
+      2 => 'tablet:grid-col-4',
+      3 => 'tablet:grid-col-6',
+      4 => 'tablet:grid-col-8',
+      5 => 'tablet:grid-col-10',
     ],
     12 => [
-      1 => 'usa-width-one-twelfth',
-      2 => 'usa-width-one-sixth',
-      3 => 'usa-width-one-fourth',
-      4 => 'usa-width-one-third',
-      5 => 'usa-width-five-twelfths',
-      6 => 'usa-width-one-half',
-      7 => 'usa-width-seven-twelfths',
-      8 => 'usa-width-two-thirds',
-      9 => 'usa-width-three-fourths',
-      10 => 'usa-width-five-sixths',
-      11 => 'usa-width-eleven-twelfths',
+      1 => 'tablet:grid-col-1',
+      2 => 'tablet:grid-col-2',
+      3 => 'tablet:grid-col-3',
+      4 => 'tablet:grid-col-4',
+      5 => 'tablet:grid-col-5',
+      6 => 'tablet:grid-col-6',
+      7 => 'tablet:grid-col-7',
+      8 => 'tablet:grid-col-8',
+      9 => 'tablet:grid-col-9',
+      10 => 'tablet:grid-col-10',
+      11 => 'tablet:grid-col-11',
     ],
   ];
   if (!empty($classes[$denominator][$numerator])) {
     return $classes[$denominator][$numerator];
   }
   // If still here, it is some weird non-supported fraction.
-  return 'usa-width-one-whole';
+  return 'tablet:grid-col-12';
 }
 
 /**
@@ -115,7 +115,7 @@ function uswds_paragraphs_preprocess_paragraph__uswds_text(&$variables) {
   if ($parent && 'paragraph' != $parent->getEntityTypeId()) {
     // If this text paragraph is not in an entity, we'll need
     // to add the grid class.
-    $variables['uswds_grid_class'] = 'usa-grid-full';
+    $variables['uswds_grid_class'] = 'grid-container padding-x-0';
   }
 }
 
@@ -123,7 +123,7 @@ function uswds_paragraphs_preprocess_paragraph__uswds_text(&$variables) {
  * Implements hook_preprocess_paragraph__uswds_grid().
  */
 function uswds_paragraphs_preprocess_paragraph__uswds_grid(&$variables) {
-  $variables['uswds_grid_class'] = 'usa-grid-full';
+  $variables['uswds_grid_class'] = 'grid-row';
   $grid = $variables['paragraph'];
   $max_columns = (int)$grid->field_uswds_grid_column_width->value;
   $columns = $grid->field_uswds_paragraphs->referencedEntities();
@@ -158,7 +158,7 @@ function uswds_paragraphs_preprocess_paragraph__uswds_grid(&$variables) {
  */
 function uswds_paragraphs_preprocess_paragraph__uswds_column(&$variables) {
   if (empty($variables['elements']['#uswds_width_class'])) {
-    $variables['elements']['#uswds_width_class'] = 'usa-width-one-whole';
+    $variables['elements']['#uswds_width_class'] = 'tablet:grid-col-12';
   }
   $width_class = $variables['elements']['#uswds_width_class'];
   $variables['attributes']['class'][] = $width_class;
@@ -169,7 +169,7 @@ function uswds_paragraphs_preprocess_paragraph__uswds_column(&$variables) {
  */
 function uswds_paragraphs_preprocess_paragraph__uswds_hero(&$variables) {
   // The "uswds_hero" Paragraph type should always be on the grid.
-  $variables['uswds_grid_class'] = 'usa-grid';
+  $variables['uswds_grid_class'] = 'grid-container';
   // We also need to set the inline CSS for the image.
   if (!$variables['paragraph']->field_uswds_image->isEmpty()) {
     // Make sure we have a unique ID, in case of multiple Heroes.
@@ -194,7 +194,7 @@ function uswds_paragraphs_preprocess_paragraph__uswds_hero(&$variables) {
 function uswds_paragraphs_preprocess_paragraph__uswds_media_block(&$variables) {
   // We need to set the media_block class here, because we side-step the field
   // template in paragraph--uswds-media-block.html.twig.
-  $variables['content']['field_uswds_image'][0]['#item_attributes']['class'][] = 'usa-media_block-img';
+  $variables['content']['field_uswds_image'][0]['#item_attributes']['class'][] = 'usa-media-block__img';
 }
 
 /**
@@ -203,7 +203,7 @@ function uswds_paragraphs_preprocess_paragraph__uswds_media_block(&$variables) {
 function uswds_paragraphs_preprocess_paragraph__uswds_graphic_list(&$variables) {
 
   // Set the grid always.
-  $variables['uswds_grid_class'] = 'usa-grid-full';
+  $variables['uswds_grid_class'] = '.grid-container.padding-x-0';
 
   // Here we split the media blocks into pairs.
   $pairs = [];
@@ -237,7 +237,7 @@ function uswds_paragraphs_preprocess_paragraph__uswds_accordion(&$variables) {
   }
 
   if ($accordion->field_uswds_accordion_bordered->value) {
-    $variables['attributes']['class'][] = 'usa-accordion-bordered';
+    $variables['attributes']['class'][] = 'usa-accordion.usa-accordion--bordered';
   }
   else {
     $variables['attributes']['class'][] = 'usa-accordion';
@@ -271,7 +271,7 @@ function uswds_paragraphs_preprocess_field__paragraph__field_uswds_link__uswds_h
  * Implements hook_preprocess_field__paragraph__field_uswds_button__uswds_hero().
  */
 function uswds_paragraphs_preprocess_field__paragraph__field_uswds_button__uswds_hero(&$variables) {
-  $classes['class'] = ['usa-button', 'usa-button-big', 'usa-button-secondary'];
+  $classes['class'] = ['usa-button', 'usa-button-big'];
   $variables['items'][0]['content']['#url']->setOption('attributes', $classes);
 }
 
